tools/github-issues/Azure.Sdk.Tools.GitHubIssues/Reports/ReportExecutionContext.cs

```diff
@@ -0,0 +1,30 @@
+﻿using Azure.Sdk.Tools.GitHubIssues.Services.Configuration;
+using Microsoft.Extensions.Logging;
+using Octokit;
+using System;
+using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+using System.Text;
+
+namespace Azure.Sdk.Tools.GitHubIssues.Reports
+{
+    public class ReportExecutionContext
+    {
+        public ReportExecutionContext(ILogger log, string fromAddress, string sendGridToken, string gitHubPersonalAccessToken, IEnumerable<RepositoryConfiguration> repositoryConfigurations, GitHubClient gitHubClient)
+        {
+            this.Log = log;
+            this.FromAddress = fromAddress;
+            this.SendGridToken = sendGridToken;
+            this.GitHubPersonalAccessToken = gitHubPersonalAccessToken;
+            this.RepositoryConfigurations = repositoryConfigurations;
+            this.GitHubClient = gitHubClient;
+        }
+
+        public ILogger Log { get; private set; }
+        public string FromAddress { get; private set; }
+        public string SendGridToken { get; private set; }
+        public string GitHubPersonalAccessToken { get; private set; }
```

Comment:
Do the tokens need to be available as properties here?

Reply:
Without them each report would need to take a dependency on SecretClient and query it directly. This seemed like a cleaner way to provide those values to each report.